state-chain/pallets/cf-trading-strategy/src/tests.rs

```diff
@@ -961,3 +1041,285 @@ mod safe_mode {
+			min_sell_tick in -5..5_i32,
+			max_sell_tick in 5..10_i32,
+		) {
+			let expected_orders = TradingStrategyPallet::inventory_base_strategy_logic(
+				AssetAmount::from(base_amount),
+				AssetAmount::from(quote_amount),
+				min_buy_tick,
+				max_buy_tick,
+				min_sell_tick,
+				max_sell_tick,
+			);
+
+			// Only doing basic checks here. We rely on the function's built in sanity checks.
+			assert!(!expected_orders.base.is_empty());
+			assert!(!expected_orders.quote.is_empty());
+		}
+	}
+
+	#[test]
+	fn logic_test() {
+		#[track_caller]
+		fn test_logic(
+			base_amount: AssetAmount,
+			quote_amount: AssetAmount,
+			min_buy_tick: Tick,
+			max_buy_tick: Tick,
+			min_sell_tick: Tick,
+			max_sell_tick: Tick,
+			expected_orders: LimitOrders,
+		) {
+			assert_eq!(
+				TradingStrategyPallet::inventory_base_strategy_logic(
+					base_amount,
+					quote_amount,
+					min_buy_tick,
+					max_buy_tick,
+					min_sell_tick,
+					max_sell_tick,
+				),
+				expected_orders,
+			);
+		}
+
+		// 50/50 split exactly
+		test_logic(
+			1000,
+			1000,
+			-10,
+			0,
+			0,
+			10,
+			LimitOrders {
+				// We expect one order each side at the average tick of each range
+				base: BTreeMap::from_iter([(5, (1, 1000))]),
+				quote: BTreeMap::from_iter([(-5, (1, 1000))]),
+			},
+		);
+
+		// 24/76 split
+		test_logic(
+			2400,
+			7600,
+			-10,
+			0,
+			0,
+			10,
+			LimitOrders {
+				// One somewhat defensive order
+				base: BTreeMap::from_iter([(8, (0, 2400))]),
+				// One average and one somewhat aggressive order
+				quote: BTreeMap::from_iter([(-5, (1, 5000)), (-2, (0, 2600))]),
+			},
+		);
+		// 76/24 split
+		test_logic(
+			7600,
+			2400,
+			-10,
+			0,
+			0,
+			10,
+			LimitOrders {
+				base: BTreeMap::from_iter([(5, (1, 5000)), (2, (0, 2600))]),
+				quote: BTreeMap::from_iter([(-8, (0, 2400))]),
+			},
+		);
+
+		// 1/99 split Asymmetric
+		test_logic(
+			10,
+			990,
+			-5,
+			0,
+			0,
+			10,
+			LimitOrders {
+				// One max defensive order
+				base: BTreeMap::from_iter([(10, (0, 10))]),
+				// One average and one max aggressive order
+				quote: BTreeMap::from_iter([(-3, (1, 500)), (0, (0, 490))]),
+			},
+		);
+
+		// Overlapping tick ranges
+		test_logic(
+			2400,
+			7600,
+			1,
+			3,
+			0,
+			5,
+			LimitOrders {
+				base: BTreeMap::from_iter([(4, (0, 2400))]),
+				quote: BTreeMap::from_iter([(2, (1, 5000)), (3, (0, 2600))]),
+			},
+		);
+
+		// Only a single tick range
+		test_logic(
+			2400,
+			7600,
+			-1,
+			0,
+			0,
+			1,
+			LimitOrders {
+				// Should always round defensively
+				base: BTreeMap::from_iter([(1, (0, 2400))]),
+				quote: BTreeMap::from_iter([(-1, (1, 5000)), (0, (0, 2600))]),
+			},
+		);
+
+		// Completely one sided
+		test_logic(
+			2000,
+			0,
+			-10,
+			0,
+			0,
+			10,
+			LimitOrders {
+				base: BTreeMap::from_iter([(5, (1, 1000)), (0, (0, 1000))]),
+				quote: BTreeMap::from_iter([]),
+			},
+		)
+	}
+
+	#[test]
+	fn test_update_threshold() {
```

Comment:
I don't think updating thresholds is the focus of this test. I would say this is more about updating orders when the threshold is reached?